models/repo.go

```diff
@@ -1275,6 +1275,20 @@ func updateRepository(e Engine, repo *Repository, visibilityChanged bool) (err e
 			}
 		}
 
+		// Create/Remove git-daemon-export-ok for git-daemon...
+		daemonExportFile := path.Join(repo.RepoPath(), `git-daemon-export-ok`)
+		if repo.IsPrivate {
+			if err = os.Remove(daemonExportFile); err != nil {
```

Comment:
As for the whole "it throws errors if it already exists or is already removed", people shouldn't change files manually (why I'm adding this :trollface: ) so it shouldn't happen and therefore isn't realyl an issue... it's might throw errors on the first update, but you can just ignore those errors... Just my 2 cents